packages/client/src/Authentication.ts

```diff
@@ -14,15 +14,15 @@ export type SignerWithProvider = AbstractSigner<Provider>
 export interface Authentication {
     // always in lowercase
     getAddress: () => Promise<EthereumAddress>
-    createMessageSignature: (payload: Uint8Array) => Promise<Uint8Array>
+    signWithWallet: (payload: Uint8Array) => Promise<Uint8Array>
```

Comment:
Maybe the `withWallet` suffix is not important as there is no other counterpart to distinguish from?  

Also, would it make just expose one `getSigner` function instead of exposing `getStreamRegistryChainSigner` and `signWithWallet`?

```suggestion
    signWithWallet: (payload: Uint8Array) => Promise<Uint8Array>
```